front/src/modules/pathfinding/utils.ts

```diff
@@ -176,3 +176,21 @@ export const isStation = (chCode: string): boolean =>
   chCode === 'BV' || chCode === '00' || chCode === '';
 
 export const isPathStepInvalid = (step: PathStep | null): boolean => step?.isInvalid || false;
+
+export function mergeDuplicatedOperationalPoints<T>(
+  operationalPoints: T[],
+  getDuplicatedOP: (operationalPoint: T) => string,
+  mergeOP: (previousOperationalPoint: T, nextOperationalPoint: T) => void
+): T[] {
+  return Object.values(
+    operationalPoints.reduce<Record<string, T>>((mergedOperationalPoints, operationalPoint) => {
+      const key = getDuplicatedOP(operationalPoint);
```

Comment:
Not sure this is the right name for the function, it just buils a key for an op

Reply:
done